### PR TITLE
[Phi] move InferShape for truncated_gaussian_random and gaussian_random

### PR DESCRIPTION
**Pull request by windstamp**
### PR types

Function optimization
### PR changes

OPs
### Describe

[Phi] move InferShape for truncated_gaussian_random and gaussian_random

### CODE REVIEW

File: paddle/phi/kernels/truncated_gaussian_random_kernel.h [hunk 1]
```diff
@@ -21,6 +21,9 @@
 #include "paddle/phi/core/dense_tensor.h"
 #include "paddle/phi/core/device_context.h"
 
+#include "paddle/phi/infermeta/nullary.h"
+#include "paddle/phi/kernels/empty_kernel.h"
```
Comment (zyfncg) [inline paddle/phi/kernels/truncated_gaussian_random_kernel.h]: 为啥include了empty_kernel.h？

**Review by zyfncg — DISMISSED** [review]

File: paddle/phi/infermeta/nullary.h [hunk 2]
```diff
@@ -40,4 +40,18 @@ void EyeInferMeta(int64_t num_rows,
...
                   MetaTensor* out);
 
+void TruncatedGaussianRandomInferMeta(const std::vector<int>& shape,
+                                      float mean,
+                                      float std,
+                                      int seed,
+                                      DataType dtype,
+                                      MetaTensor* out);
+
+void GaussianRandomInferMeta(const ScalarArray& shape,
```
Comment (YuanRisheng) [inline paddle/phi/infermeta/nullary.h:50]: 这里不是vector<int>吗，我看下边的kernel全部替换成了vector<int>

Comment (windstamp) [reply paddle/phi/infermeta/nullary.h:50]: truncated_gaussian_random 是 std::vector<int>，gaussian_random 还是 ScalarArray。

File: paddle/phi/kernels/truncated_gaussian_random_kernel.h (hunk 1 above)
Comment (windstamp) [reply paddle/phi/kernels/truncated_gaussian_random_kernel.h]: 删掉了，thanks。

**Review by zyfncg — APPROVED** [review]

**Review by YuanRisheng — APPROVED** [review]

**Pull request merged**